src/main/java/com/seals/camplanner/event/controllers/EventController.java

```diff
@@ -12,25 +15,44 @@
 @RequiredArgsConstructor
 public class EventController {
     private final EventService eventService;
+    private final LocationService locationService;
+    private final ModelMapper modelMapper;
 
     @GetMapping("/event")
-    public List<Event> getEvents() {
-        return eventService.findAll();
+    public List<EventDto> getEvents() {
+        return toEventDtoList(eventService.findAll());
     }
 
     //TODO add treatment for the "location" parameter
     @PostMapping("/event")
-    public Event saveEvent(@RequestBody EventDto event) {
-        return eventService.save(event);
+    public EventDto saveEvent(@RequestBody EventDto event) {
+        return toDto(eventService.save(fromDto(event)));
     }
 
     @GetMapping("/event/{id}")
-    public Event getEvent(@PathVariable("id") Long id) {
-        return eventService.find(id);
+    public EventDto getEvent(@PathVariable("id") Long id) {
+        return toDto(eventService.find(id));
     }
 
     @DeleteMapping("/event/{id}")
     public void deleteEvent(@PathVariable("id") Long id) {
         eventService.delete(id);
     }
+
+    private EventDto toDto(Event event) {
+        EventDto eventDto = modelMapper.map(event, EventDto.class);
+        eventDto.setLocationId(event.getLocation().getId());
+        return eventDto;
+    }
+
+    private Event fromDto(EventDto eventDto) {
+        Event event = modelMapper.map(eventDto, Event.class);
+        Location location = locationService.find(eventDto.getLocationId());
+        event.setLocation(location);
+        return event;
```

Comment:
I don't feel like the Controller should be the owner of the transform methods.

I think I'd rather do something like
```
LocationDTO locationDto = LocationDTO.of(location)
// and similarly
Location location = Location.fromDto(locationDTO)
```
To transform from and to DTOs.

Another options would be to use the Controller (or some other Singleton) initialization to configure the ModelMapper with appropriate [Converters](http://modelmapper.org/user-manual/converters/) and [Providers](http://modelmapper.org/user-manual/providers/) so it can do the whole conversion by itself. This also seems to be a good option since when can access the ModelMapper at anytime using Spring, we could probably have a mix of both options.